src/data_source/storage/sql/queries.rs

```diff
@@ -94,7 +94,12 @@ impl<'q> QueryBuilder<'q> {
         self.arguments.add(arg).map_err(|err| QueryError::Error {
             message: format!("{err:#}"),
         })?;
-        Ok(format!("${}", self.arguments.len()))
+
+        if cfg!(feature = "embedded-db") {
+            Ok("?".to_string())
```

Comment:
I believe sqlite does allow the `$` syntax in query params. I don't know if it respects the indexing or just converts it to `?` internally, but I think we might not need this `if`

Reply:
you are right!